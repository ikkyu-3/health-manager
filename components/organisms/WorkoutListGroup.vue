<template>
  <v-list-item-group>
    <v-subheader>{{ subHeader }}</v-subheader>
    <template v-for="(item, index) in items">
      <workout-list-item
        :key="index"
        :disabled="item.disabled"
        :index="item.index"
        :name="item.name"
        @click="itemClick(item.index, item.name)"
      />
    </template>
  </v-list-item-group>
</template>

<script lang="ts">
import { createComponent } from '@vue/composition-api'
import WorkoutListItem from '@/components/molecules/list-items/WorkoutListItem.vue'

export default createComponent({
  components: { WorkoutListItem },
  props: {
    subHeader: { type: String, require: true },
    items: { type: Array, require: true },
    itemClick: { type: Function, require: true }
  }
})
</script>
