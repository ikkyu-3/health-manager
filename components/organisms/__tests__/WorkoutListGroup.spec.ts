import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import WorkoutListGroup from '@/components/organisms/WorkoutListGroup.vue'

const itemClick = jest.fn()
const options = {
  localVue: createLocalVue(),
  vuetify: new Vuetify(),
  propsData: {
    subHeader: 'Sub Header',
    items: [
      {
        disabled: false,
        index: '1',
        name: 'item'
      }
    ],
    itemClick
  }
}

describe('organisms/WorkoutListGroup.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(WorkoutListGroup, { ...options })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Listをクリックすると、引数にindexとnameが渡される', () => {
    const wrapper = mount(WorkoutListGroup, { ...options })
    const item = wrapper.find('div[role="listitem"]')
    item.trigger('click')
    expect(itemClick.mock.calls[0]).toEqual(['1', 'item'])
  })
})
