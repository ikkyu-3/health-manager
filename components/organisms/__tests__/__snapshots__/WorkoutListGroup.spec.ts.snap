// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/WorkoutListGroup.vue snapshot 1`] = `"<div role=\\"listbox\\" class=\\"v-item-group theme--light v-list-item-group\\"><div class=\\"v-subheader theme--light\\">Sub Title</div> <div tabindex=\\"0\\" role=\\"listitem\\" aria-selected=\\"false\\" class=\\"v-list-item v-list-item--link theme--light\\"><div class=\\"v-list-item__icon\\"><div class=\\"icon active\\"><span>1</span></div></div> <div class=\\"v-list-item__content\\"><div class=\\"v-list-item__title\\">item</div></div></div></div>"`;
